api/modules/currency/views.py

```diff
@@ -33,3 +35,51 @@ def get_currency_exchange_rate(request, source_currency_code, target_currency_co
         return Response(exception_message, status=status.HTTP_503_SERVICE_UNAVAILABLE)
 
     return Response(response.to_json())
+
+
+@api_view(['GET'])
+def get_all_currency_exchange_rate(request, start_date, end_date, source_currency_code, target_currency_code):
+    """
+    Return currency exchange rates list between 2 dates
+    :param request:
+    :param start date:
+    :param end date:
+    :param source_currency_code:
+    :param target_currency_code:
+    :return 400 response if dates are incorrect
+    :return: 503 if Free Currency Converter api fails
+    :return: 200 successful
+    """
+    currency_list = []
+    y_start = start_date.split('-')[0]
+    m_start = start_date.split('-')[1]
+    d_start = start_date.split('-')[2]
+    start = datetime.datetime.strptime(y_start+m_start+d_start, '%d%m%Y').date()
+    y_end = end_date.split('-')[0]
+    m_end = end_date.split('-')[1]
+    d_end = end_date.split('-')[2]
+    end = datetime.datetime.strptime(y_end+m_end+d_end, '%d%m%Y').date()
+    if end < start:
+        error_message = "End Date is before Start Date"
```

Comment:
`End date should be before start date`

Reply:
I'm sorry how should end date be before start date